### PR TITLE
[qa_automation] Add kotd module

### PR DESCRIPTION
**Pull request by jtzhao**
Add kotd(kernel of the day) module to help install kotd kernels and reboot system.

### CODE REVIEW

**Comment by richiejp** [comment]
How is this test module ran?

**Comment by coolo** [comment]
This looks like a basetest - or a library, not a test.

**Comment by richiejp** [comment]
That is probably OK, I can use it in ltp_install, but on the other hand it might be better off as a self contained test module.

Also I think this should be in the kernel folder with the other new LTP stuff. Thanks.

**Comment by jtzhao** [comment]
@richiejp This is a library. I'll move it to the kernel folder and add more comments for the functions.

File: tests/qa_automation/kotd.pm [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    if ($output =~ /(?<='zypper install )([^']+)/) {
+        assert_script_run("zypper -n install $1", 1200);
+    }
+    else {
+        die "Failed to install kernel of the day";
+    }
+}
+
+sub kotd_reboot {
+    my $self = shift;
```
Comment (okurz) [inline tests/qa_automation/kotd.pm]: delete unused $self

File: tests/qa_automation/kotd.pm [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+use testapi;
+
+sub kotd_addrepo {
+    my ($self, $url) = @_;
+    assert_script_run("zypper --no-gpg-check -n ar -f '$url' kotd", 600);
+    assert_script_run("zypper --gpg-auto-import-keys ref",          600);
+}
+
+sub kotd_install {
+    my $self   = shift;
```
Comment (okurz) [inline tests/qa_automation/kotd.pm]: delete unused $self

File: tests/qa_automation/kotd.pm [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    my $output = script_output("zypper -n up kernel-default");
+    if ($output =~ /(?<='zypper install )([^']+)/) {
+        assert_script_run("zypper -n install $1", 1200);
+    }
+    else {
+        die "Failed to install kernel of the day";
+    }
+}
```
Comment (okurz) [inline tests/qa_automation/kotd.pm]: why not just `zypper_call('up kernel-default')`?

File: tests/kernel/install_kotd.pm [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    my $url     = "http://download.suse.de/ibs/Devel:/Kernel:/$release/standard/";
+    zypper_call("--no-gpg-check ar -f '$url' kotd", timeout => 600);
+    zypper_call("--gpg-auto-import-keys ref",       timeout => 1200);
+}
+
+# Install kotd kernel
+sub kotd_install {
+    my $output = script_output("zypper -n up kernel-default");
+    if ($output =~ /(?<='zypper install )([^']+)/) {
+        zypper_call("install $1", timeout => 1200);
```
Comment (jtzhao) [inline tests/kernel/install_kotd.pm]: "zypper in kernel-default" will fail, saying the new kernel is from another vendor. Then, zypper will show the correct command to install it. That's why I used script_output to capture the output.

Comment (coolo) [inline tests/kernel/install_kotd.pm]: try in --from kotd

Comment (coolo) [inline tests/kernel/install_kotd.pm]:
```
cleopatra:~ # LC_ALL=C zypper -n in --from 3 perl-Module-Runtime
Loading repository data...
Reading installed packages...
Resolving package dependencies...

The following package is going to be upgraded:
  perl-Module-Runtime

The following package is going to change vendor:
  perl-Module-Runtime  openSUSE -> obs://build.opensuse.org/devel:languages:perl


1 package to upgrade, 1 to change vendor.
Overall download size: 23.9 KiB. Already cached: 0 B. No additional space will be used or freed after the operation.
Continue? [y/n/? shows all options] (y): y
Retrieving package perl-Module-Runtime-0.014-19.8.noarch

**Comment by jtzhao** [comment]
@okurz Hi, any idea why the checks failed? It doesn't seem like a coding style problem:
tools/tidy --check
cannot find file given with -pro=.../.perltidyrc: No such file or directory

**Comment by okurz** [comment]
please rebase on current master and make sure your "os-autoinst" subfolder is up to date to use an updated style check.

**Pull request closed without merging**